src/workerd/jsg/util.c++

```diff
@@ -90,9 +92,53 @@ kj::String typeName(const std::type_info& type) {
   return kj::mv(result);
 }
 
+namespace {
+
+// For internal errors, we generate an ID to include when rendering user-facing "internal error"
+// exceptions and writing internal exception logs, to make it easier to search for logs
+// corresponding to "internal error" exceptions reported by users.
+//
+// We'll use an ID of 24 base-32 encoded characters, just because its relatively simple to
+// generate from random bytes.  This should give us a value with 120 bits of uniqueness, which is
+// about as good as a UUID.
+//
+// (We're not using base-64 encoding to avoid issues with case insensitive search, as well as
+// ensuring that the id is easy to select and copy via double-clicking.)
+using InternalErrorId = kj::FixedArray<char, 24>;
+
+constexpr char BASE32_DIGITS[] = "0123456789abcdefghijklmnopqrstuv";
+
+InternalErrorId makeInternalErrorId() {
+  InternalErrorId id;
+  if (isPredictableModeForTest()) {
+    // In testing mode, use content that generates a "0123456789abcdefghijklm" ID:
+    for (auto i: kj::indices(id)) {
+      id[i] = i;
+    }
+  } else {
+    KJ_ASSERT(RAND_bytes(id.asPtr().asBytes().begin(), id.size()) == 1);
+  }
+  for (auto i: kj::indices(id)) {
+    id[i] = BASE32_DIGITS[static_cast<unsigned char>(id[i]) % 32];
+  }
+  return id;
+}
+
+kj::String renderInternalError(InternalErrorId& internalErrorId) {
+  // TODO(now): put "internal error" change behind autogate or compatibility flag?
```

Comment:
Do we make any guarantees about the error messages we send back?  If we don't, and I don't think we do, I think it's fine to change what we return.  (And hope that we don't lose this round with Hyrum's Law...)

Reply:
Heh... yeah, that's kind of what I was wondering.  I do know that the custom `.retryable` and `.overloaded` exception properties were introduced in part to reduce the necessity of inspecting exception strings, as was observed in some internal workers.

But IIRC, those workers didn't look for bare `internal error` messages...  which I guess makes sense, since they should be pretty rare, since they go to internal logs.

Reply:
Hmm, anyone who depends on string-matching `"internal error"` is on extremely precarious ground since it always represents some sort of bug in our system (where often the bug is simply "we forgot to assign a better error message")...

I could imagine that someone is doing it, but what I'm not sure of is whether it's likely that we'll outright break them by changing the message. More likely we'll just slightly degrade their own error handling?

I'm inclined to say we don't need a compat flag here... I think...